### PR TITLE
Use primitive types

### PR DESCRIPTION
**Pull request by solcott**
Use primitive types when possible.  This fixes NullPointerExceptions when properties that are primitive types are not in the supplied json

### CODE REVIEW

File: src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java [hunk 1]
```diff
@@ -347,6 +359,34 @@ public MethodSpec createReadMethod(ClassName className,
...
+                valueString = valueString + "f";
+                break;
+              case "long":
+                valueString = valueString + "L";
+            }
+          }
+        }
+      }
+
+    } catch (Exception e) {
```
Comment (JakeWharton) [inline src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java]: Why is this here with such a wide exception type?

File: src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java [hunk 2]
```diff
@@ -347,6 +359,34 @@ public MethodSpec createReadMethod(ClassName className,
...
+          if (!Strings.isNullOrEmpty(valueString)) {
+            switch (type.toString()) {
+              case "double":
+                valueString = valueString + "d";
+                break;
+              case "float":
+                valueString = valueString + "f";
+                break;
+              case "long":
+                valueString = valueString + "L";
```
Comment (JakeWharton) [inline src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java]: All `case` branches should include a `break;`

Comment (solcott) [inline src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java]: Fixed

File: src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java (hunk 1 above)
Comment (solcott) [inline src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java]: Changed this to just catch ClassNotFoundException

File: src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java [hunk 3]
```diff
@@ -296,8 +299,17 @@ public MethodSpec createReadMethod(ClassName className,
     for (Property prop : adapters.keySet()) {
       FieldSpec field = FieldSpec.builder(prop.type, prop.name).build();
       fields.put(prop, field);
-
-      readMethod.addStatement("$T $N = null", field.type.isPrimitive() ? field.type.box() : field.type, field);
+      if(field.type.isPrimitive()){
```
Comment (rharter) [inline src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java]: Please fix up the code format.  There seem to be inconsistent whitespace around the if/else blocks.

Comment (solcott) [inline src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java]: Fixed

**Comment by rharter** [comment]
Looks good. :+1:
